cmd/pipeline/list.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	contextCMD "github.com/okteto/okteto/cmd/context"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/constants"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"io"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"strings"
+	"text/tabwriter"
+)
+
+type listFlags struct {
+	labels    []string
+	namespace string
+	output    string
+}
+
+type pipelineListItem struct {
+	Name       string   `json:"name" yaml:"name"`
+	Status     string   `json:"status" yaml:"status"`
+	Repository string   `json:"repository" yaml:"repository"`
+	Branch     string   `json:"branch" yaml:"branch"`
+	Labels     []string `json:"labels" yaml:"labels"`
+}
+
+func list(ctx context.Context) *cobra.Command {
+	flags := &listFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all okteto pipelines",
+		Args:  utils.NoArgsAccepted(""),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if flags.namespace == "" {
```

Comment:
Maybe we can wrap all the context initialisation in order to make it testable.

Reply:
I made some changes now, not perfect but still it's an improvement. Let me know if you have feedback on that